main/lsp/requests/document_symbol.cc

```diff
@@ -79,6 +81,16 @@ bool DocumentSymbolTask::isDelayable() const {
     return true;
 }
 
+bool isOwnerInTheSameFile(const core::GlobalState &gs, core::SymbolRef ref, core::FileRef fref) {
+    auto &locs = ref.owner(gs).locs(gs);
+
+    return absl::c_any_of(locs, [fref](auto &loc) { return loc.file() == fref; });
```

Comment:
Nit: Isn't this equivalent?

```suggestion
    return isRefInFile(gs, ref.owner(gs), fref);
```

(You'll need to move `isRefInFile` above `isOwnerInTheSameFile`.)